### PR TITLE
Make flatpak plugins optional

### PR DESCRIPTION
**Pull request by csomh**
Flatpak related plugin `pre_resolve_module_compose` pulls in
`modulemd` and `pdc_client` as a dependency. Through
`pre_resolve_module_compose` this gets propagated to
`exit_koji_promote` and `exit_koji_import`, which will render
buildroots having koji related plugins installed - but missing
these two flatpak related dependencies - unusable.

This change will catch import errors for get_flatpak_source_info()
and define the function to return None.

This is safe as the update of extra['image'] will only happen
if get_flatpak_source_info() returns *not None* (which is the case
when flatpak related plugins are disabled).

Signed-off-by: Hunor Csomortáni <csomh@redhat.com>

### CODE REVIEW

**Comment by csomh** [comment]
@twaugh @lcarva ptal

**Comment by vrutkovs** [comment]
Weird, I tried several buildroots without `modulemd` and they worked.

A proper solution would be moving `get_compose_info/set_compose_info` into `utils.py` though

**Comment by csomh** [comment]
@vrutkovs that would make sense, but these two functions seem to be dependent on `pre_resolve_module_compose`:

```python
def get_compose_info(workflow):
    key = ResolveModuleComposePlugin.key
...
```
so maybe they are intended to be used only in relation with this plugin.

`key` could be made an argument of course, but that would not help with breaking up dependencies with the koji plugins.

**Comment by vrutkovs** [comment]
All keys would ideally be stored in `constants.py`, as we do in many places.

In any case, the coverage has decreased, we should consider adding some tests here

**Comment by twaugh** [comment]
@vrutkovs, have any suggestions for how to add unit testing for it?

**Comment by vrutkovs** [comment]
Need to trigger this in centos tests:
```
if [[ $OS == "fedora" ]]; then
  $RUN $PIP install -r requirements-flatpak.txt
fi
```

File: tests/plugins/test_flatpak_create_oci.py [hunk 1]
```diff
@@ -634,6 +640,8 @@ def get_file_perms(self, path):
             return '-{:05o}'.format(stat.S_IMODE(mode))
 
 
+@pytest.mark.skipif(not MODULEMD_AVAILABLE,
```
[the remark this answers is not in the document]
Comment (vrutkovs) [reply tests/plugins/test_flatpak_create_oci.py]: Needs `# noqa` comment

**Comment by csomh** [comment]
Tests are fixed and coverage is back. Thanks @vrutkovs for the tip!

Let me know if I should squash the 2 commits.

**Comment by vrutkovs** [comment]
Release notes updated

**Pull request merged**